jerry-core/jrt/jrt.h

```diff
@@ -86,9 +86,9 @@ extern void __noreturn jerry_unimplemented (const char *, const char *, const ch
 #define JERRY_LOG(lvl, ...) \
   do \
   { \
-    if (lvl <= jerry_debug_level && jerry_log_file) \
+    if (lvl <= jerry_debug_level) \
```

Comment:
IMHO, this check should not happen here but in the port implementation (if that impl cares about it). Likewise, `jerry_debug_level` shouldn't even be defined in jerry.c. Everything `JERRY_LOG` should do is just pass things on to `jerry_port_log` implemented in the port for it to decide what to do with it (as `lvl` is passed on as well).

In some sense, it should work similarly to the "fatal" logic. As far ar jerry-core is concerned, there is only a single `jerry_port_fatal` function to be called. However, the default port implementation adds some extras to it, namely the "abort_on_fail" functionality. However, that's a port-specific thing. Therefore, the `static bool abort_on_fail` variable is defined in jerry-port-default-fatal.c, and accessor methods `jerry_port_default_set_abort_on_fail` and `jerry_port_default_is_abort_on_fail` in jerry-port-default.h.

I can imagine a similar scenario with the logs. There could be default port-specific accessors, e.g., `jerry_port_default_set_log_level` and `jerry_port_default_get_log_level` in jerry-port-default.h and their implementation (along with any variables needed for that logic, say `debug_level` or `log_level`) in jerry-port-default-io.c. And then the line `(void) level; /* default port implementation ignores the log level */` will go away (actually, the presence of this line already signals the problem that log levels are not checked at the right place, IMHO).

Additonal notes:
* `JERRY_D[D[D]]LOG` macros call `JERRY_LOG` with 1, 2, and 3 as level parameters. This does not seem right in the light of the new log level enum.
* There are still some `printf`s in the code (e.g., jrt-fatals.c, js-parser.c, js-parser-util.c, ecma-builtin-global.c, jmem-poolman.c, jmem-heap.c, js-parser-statm.c), even if mostly macro-guarded. When will these be rewritten to call the `_log` and `_console` functions?